taf/git.py

```diff
@@ -781,26 +781,29 @@ def is_branch_with_unpushed_commits(self, branch_name):
                 message="pygit repository could not be instantiated.",
             )
 
-        # Ensure the branch exists locally.
-        local_branch_ref = f"refs/heads/{branch_name}"
-        if local_branch_ref not in repo.references:
-            return False
+        local_branch = repo.branches.get(branch_name)
 
-        # Ensure the branch has an upstream.
-        try:
-            upstream_branch_ref = repo.lookup_branch(branch_name).upstream.name
-        except ValueError:
-            # not upstream yet
+        upstream_full_name = local_branch.upstream_name
+        if not upstream_full_name:
+            # no upstream branch - not pushed
             return True
+        parts = upstream_full_name.split("/")
+        upstream_name = "/".join(parts[2:])
```

Comment:
Why does `upstream_name` start with '/'?

Reply:
It's `refs/remotes/origin/branch_name`, so we remove the first 2 parts